### DISCUSSION TITLE
GHCR images not visible to organization owner

### DISCUSSION THREAD

**Discussion by lizrice** [community/community#27191] [Actions]
<p>I would expect that as an organization owner, I’d be able to see all container images under the organization, and I’d be able to change their visibility settings. That doesn’t seem to be the case and I wonder if this is a bug. Here’s what happened:</p>
<ul>
<li>User A pushes a container image (e.g. <code>docker push ghcr.io/aquasecurity/trivy:latest</code>)</li>
<li>User B is an organization owner, but B can’t see this container image
<ul>
<li>it doesn’t appear in the list of packages at <a href="https://github.com/orgs/aquasecurity/packages" rel="nofollow noopener">https://github.com/orgs/aquasecurity/packages</a>
</li>
<li>if user B is logged in and tries to pull the image they get <code>Error response from daemon: pull access denied for ghcr.io/aquasecurity/trivy, repository does not exist or may require 'docker login': denied: permission_denied: read_package</code>
</li>
</ul>
</li>
<li>If user A changes the visibility of the image to public, then it’s visible to user B in the UI, and can be pulled, but under the package settings for the image, user B sees the error message “You do not have permissions to administrate this package.”</li>
</ul>
<p>Related, is there any relationship between permissions on a repo, and container image permissions with a matching name? For example, if user C is an admin on the aquasecurity/trivy repo, would you expect them to have visibility and management permissions on the <a href="http://ghcr.io/aquasecurity/trivy" rel="nofollow noopener">ghcr.io/aquasecurity/trivy</a> image? I think this would be helpful.</p>

**Comment by clarkbw** [comment]
<p>Hello! We’re working on a complete fix for this but in the meantime you’re right that org owners can’t see private containers. There will be some options here but certain scenarios require private containers to be hidden even from owners. Again we’ll be addressing the default case you’re experiencing here.</p>
<p>If you want to keep the containers private but grant access you can add User B (or a Team that contains User B) in the container settings with read/write/admin access.</p>
<p>Currently there isn’t a direct relationship between a repo and a container for access control. The Docker service <code>docker.pkg.github.com</code> worked this way and we’re going to bring the good parts of that back while leaving the fine grained options.</p>
<p>You can use Teams to create a more manageable relationship where <code>Team Admin</code> has admin access to the repo and can also be granted admin access to the Container.</p>
<p>We’ll also have some refinements to this coming up as we workout the <code>GITHUB_TOKEN</code> support in Actions.</p>

**Comment by Cian911** [comment]
<p>Hi @ <a href="https://github.community/u/clarkbw">clarkbw</a> is there any update on this?</p>

**Comment by gautamkrishnar** [comment]
<p>No update on this yet?. I faced the same problem today.</p>

**Comment by clarkbw** [comment]
<p>We have a fix in the works that allows for org admins to be able to see and admin all containers within their org.  Coming soon.</p>

**Comment by clarkbw** [comment, marked as answer]
<p>This was released a month ago <a href="https://github.blog/changelog/2020-11-16-packages-organization-admins-access-to-containers/">https://github.blog/changelog/2020-11-16-packages-organization-admins-access-to-containers/</a></p>
